manifests/init.pp

```diff
@@ -116,6 +116,12 @@
   $wait_for_pxp_agent_exit = undef,
 ) inherits ::puppet_agent::params {
 
+  if any_resources_of_type('filebucket', { path => false }) {
```

Comment:
Since this a bit of an expensive check, i think this should be done after validating that the requested version is valid and only if $aio_upgrade_require is true